lustrefs-exporter/src/jobstats.rs

```diff
@@ -153,6 +154,12 @@ pub fn jobstats_stream<R: BufRead + std::marker::Send + 'static>(
                 tracing::debug!("Unexpected error processing jobstats lines: {e}");
             };
         }
+
+        if let Some(mut child) = child {
+            if let Err(e) = child.wait() {
```

Comment:
This is going to block until the subcommand finishes, that will effectively wait out the command instead of streaming.

Reply:
This done after handling all lines, shouldn't this be OK?

Reply:
No, it's all async.

rx get's returned and nothing happens until something reads from it.

Reply:
This block is not async, I might be misunderstanding. Here is my understanding of it.

1. We spawn a blocking task that will run in background
2. The task read a line, process it and send it.
3. The send is blocking, so until the receiver read it, it will not read a new line or continue.
4. Once all lines are read, we wait for the child to exit ( this assumes the command exited after all lines were read)

So it shouldn't be an issue, when we wait for child, we have already processed and sent all lines.
No one is waiting on the blocking task too, so it exits as soon as the task return.

Reply:
Sigh, I missed that it's inside the `spawn_blocking` due to github PR formatting (sorry about that).

This will cover the case where we don't return early. where we do return early, we still have the same issue. Perhaps we should move the child.wait to a separate spawn_blocking in main.

Reply:
Posted an update that moves the child.wait into a separate spawn_blocking.